torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1882,7 +1905,13 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::broadcast(
 
   // @lint-ignore CLANGTIDY
   auto tensor = tensors.back();
-  RECORD_PARAM_COMMS(
+
+  RECORD_PARAM_COMMS_DATA(
+      static_cast<int>(
+          this->getSequenceNumberForGroup() + 1), // seq + 1 to match collective
+      reinterpret_cast<std::intptr_t>(this), // work obj ptr
```

Comment:
Confused about recording `this`. Isn't this a pointer to ProcessGroupNCCL rather than a work obj? Same applies to the other collectives

Reply:
Modified the comment, the code is correct, but the comment was wrong. Should be the process group object.